vm/package/execute.rs

```diff
@@ -0,0 +1,342 @@
+        let (execution_response, execution) = process.execute::<A, R>(authorization, rng)?;
+
+        // Execute the additional fee.
+        let (_, additional_fee) = process.execute_additional_fee::<A, _>(private_key, credits, 1, rng)?;
+
+        // TODO: the additional_fee should not be optional.
+        let execution_transaction = Transaction::from_execution(execution, Some(additional_fee))?;
+
+        let caller = self.manifest_file().development_address();
+
+        match endpoint {
+            Some(ref endpoint) => {
+                // Construct the deploy request.
+                let request = ExecutionRequest::new(execution_transaction, *caller, *program_id)?;
+                // Send the deploy request.
+                let response = request.send(endpoint)?;
+                Ok((execution_response, response.transaction))
+            }
+            None => Ok((execution_response, execution_transaction)),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    type CurrentNetwork = snarkvm_console::network::Testnet3;
+    type CurrentAleo = snarkvm_circuit::network::AleoV0;
+
+    #[test]
+    fn test_execute() {
+        // Samples a new package at a temporary directory.
+        let (directory, package) = crate::package::test_helpers::sample_package();
+
+        // Initialize caller.
+        let caller_private_key = package.manifest_file().development_private_key();
+        let caller = Address::try_from(caller_private_key).unwrap();
+        let caller_credits = Record::<CurrentNetwork, Plaintext<CurrentNetwork>>::from_str(&format!(
+            "{{ owner: {caller}.private, gates: 5u64.private, _nonce: 0group.public }}"
+        ))
+        .unwrap();
+
+        // Initialize the function inputs.
+        let r0 = Value::<CurrentNetwork>::from_str(&caller.to_string()).unwrap();
+        let r1 = Value::<CurrentNetwork>::from_str("99u64").unwrap();
+
+        // Deploy the package.
+        let (execution_response, execution_transaction) = package
+            .execute::<CurrentAleo, _>(
+                None,
+                caller_private_key,
+                caller_credits,
+                Identifier::from_str("mint").unwrap(),
+                &[r0, r1],
+                &mut rand::thread_rng(),
+            )
+            .unwrap();
+        if let Transaction::Execute(_, execution, Some(_)) = execution_transaction {
+            // Ensure the deployment edition matches.
+            assert_eq!(<CurrentNetwork as Network>::EDITION, execution.edition());
+            // Ensure that the number of outputs is correct.
+            assert_eq!(execution_response.outputs().len(), 1);
+            // Ensure that the execution output is correct.
+            assert!(execution_response.outputs().iter().all(|output| matches!(output, Value::Record(_record))));
+        }
+
+        // Proactively remove the temporary directory (to conserve space).
+        std::fs::remove_dir_all(directory).unwrap();
```

Comment:
```suggestion
```
[`tempfile::tempdir`](https://docs.rs/tempfile/3.3.0/tempfile/fn.tempdir.html) automatically removes the temporary directory on `Drop`